tests/test_http.py

```diff
@@ -182,6 +182,57 @@ def test_size_exceeded(self):
                     b"",
                 )
 
+    def test_size_check_ignore_files(self):
+        body = (
+            b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="title"\r\n\r\n'
+            + b"My First Book\r\n"
+            + b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="pdf"; filename="book.pdf"\r\n\r\n'
+            + b"FAKEPDFBYTESGOHERETHISISREALLYLONGBUTNOTUSEDTOCOMPUTETHESIZEOFTHEREQUEST"
+            + b"--BOUNDARY--"
+        )
+        request = AsgiRequest(
+            {
+                "http_version": "1.1",
+                "method": "POST",
+                "path": "/test/",
+                "headers": {
+                    "content-type": b"multipart/form-data; boundary=BOUNDARY",
+                    "content-length": str(len(body)).encode("ascii"),
+                },
+            },
+            body,
+        )
+        with override_settings(DATA_UPLOAD_MAX_MEMORY_SIZE=60):
+            request.POST
+
+    def test_size_check_ignore_files_but_honor_other_post_data(self):
+        body = (
+            b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="title"\r\n\r\n'
+            + b"My First Book\r\n"
+            + b"--BOUNDARY\r\n"
+            + b'Content-Disposition: form-data; name="pdf"; filename="book.pdf"\r\n\r\n'
+            + b"FAKEPDFBYTESGOHERETHISISREALLYLONGBUTNOTUSEDTOCOMPUTETHESIZEOFTHEREQUEST"
+            + b"--BOUNDARY--"
+        )
+        request = AsgiRequest(
+            {
+                "http_version": "1.1",
+                "method": "POST",
+                "path": "/test/",
+                "headers": {
+                    "content-type": b"multipart/form-data; boundary=BOUNDARY",
+                    "content-length": str(len(body)).encode("ascii"),
+                },
+            },
+            body,
+        )
+        with override_settings(DATA_UPLOAD_MAX_MEMORY_SIZE=1):
+            with pytest.raises(RequestDataTooBig):
+                request.POST
```

Comment:
(I'm so puzzled by the fact that this is raised during access of request.POST while the check itself is in the request constructor)

Reply:
@jpic the raise in the constructor only applies when we're not in a multipart request. The one on the test you're commenting is produced by django: https://github.com/django/django/blob/77d25dbd0f20d6a907c805ffae8aaadd87edbacf/django/http/multipartparser.py#L198

Reply:
Thanks for the heads up ! so, the tests of this patch do not actually test the change of runtime code that the patch introduces.